### PR TITLE
Convert knobs and actions to controls/args for MainQuoteSummary

### LINKED ISSUE #13175: MainQuoteSummary story: convert knobs and actions to controls / args
**Describe the bug**
As part of a push to convert all stories that use deprecated `@storybook/addon-knobs` and `@storybook/addon-actions` to use [controls](https://storybook.js.org/docs/react/essentials/controls) and [action argType annotation](https://storybook.js.org/docs/react/essentials/actions#action-argtype-annotation)

Convert `MainQuoteSummary` story in `ui/pages/swaps/main-quote-summary/main-quote-summary.stories.js` to use [controls](https://storybook.js.org/docs/react/essentials/controls) and [action argType annotation](https://storybook.js.org/docs/react/essentials/actions#action-argtype-annotation)

Read the [MetaMask extension storybook documentation guidelines](https://metamask.github.io/metamask-storybook/?path=/story/getting-started-documentation-guidelines--page) to get a better understanding of writing stories. Providing MDX documentation although helpful, is a bonus and not a requirement for this issue.

To start storybook:
```
yarn storybook
```

**Requirements**
- [ ] Story has `argTypes` that align with component api / props
- [ ] All instances of `@storybook/addon-knobs` have been removed in favour of control args
- [ ] All instances of `@storybook/addon-actions` have been removed in favour of [action argType annotation](https://storybook.js.org/docs/react/essentials/actions#action-argtype-annotation)

Any further improvements to the story interaction or documentation is a bonus

**Images**

Screen Shot 2021-12-28 at 8 15 03 AM
Screen Shot 2021-12-28 at 8 15 21 AM

### PR DESCRIPTION
**Pull request by Jordon-L**
## Explanation
This PR converts the old knobs and actions to controls/args for the MainQuoteSummary component

## More Information

Fixes #13175 

### Before

before
before2

### After

after-controls

after-readmeMDX

## Manual Testing Steps
run `yarn storybook`

## Pre-Merge Checklist

- [ ] PR template is filled out
- [ ] Manual testing complete & passed
- [ ] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** QA attention is required, "QA Board" label has been applied
- [ ] PR has been added to the appropriate release Milestone

### CODE REVIEW

**Comment by Jordon-L** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by Jordon-L** [comment]
Should I remove the controls for sourceIconUrl and destinationIconUrl?

Screenshot:
remove-control

**Review by darkwing — CHANGES_REQUESTED** [review]
Tiny change requested, otherwise beautiful!  Thank you!

File: ui/pages/swaps/main-quote-summary/main-quote-summary.js [hunk 1]
```diff
@@ -129,21 +129,54 @@ export default function MainQuoteSummary({
...
     PropTypes.instanceOf(BigNumber),
   ]).isRequired,
+
+  /**
+   * Maximum number of decimal places for the source token.
+   */   
   sourceDecimals: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+
+  /**
+   * The ticker symbol for the source token
```
[the remark this answers is not in the document]
Comment (Jordon-L) [reply ui/pages/swaps/main-quote-summary/main-quote-summary.js]: Thanks, I have added a "."

**Review by darkwing — DISMISSED** [review]

File: ui/pages/swaps/main-quote-summary/main-quote-summary.stories.js [hunk 2]
```diff
@@ -17,18 +61,11 @@ export const DefaultStory = () => {
...
-        destinationIconUrl=".storybook/images/sai.svg"
+      <MainQuoteSummary 
+        {...args}
       />
     </div>
   );
 };
 
-DefaultStory.storyName = 'Default';
+DefaultStory.storyName = 'MainQuoteSummary';
```
Comment (georgewrmarshall) [inline ui/pages/swaps/main-quote-summary/main-quote-summary.stories.js:71]: We have a convention where the first story is always named "Default" then any proceeding stories are named with more context.

```suggestion
DefaultStory.storyName = 'Default';
```

File: ui/pages/swaps/main-quote-summary/main-quote-summary.stories.js [hunk 3]
```diff
@@ -17,18 +61,11 @@ export const DefaultStory = () => {
...
-        sourceDecimals={number('sourceDecimals', 18)}
-        sourceSymbol={text('sourceSymbol', 'ETH')}
-        destinationValue={text('destinationValue', '200000000000000000')}
-        destinationDecimals={number('destinationDecimals', 18)}
-        destinationSymbol={text('destinationSymbol', 'ABC')}
-        sourceIconUrl=".storybook/images/metamark.svg"
-        destinationIconUrl=".storybook/images/sai.svg"
+      <MainQuoteSummary 
+        {...args}
       />
```
Comment (georgewrmarshall) [inline ui/pages/swaps/main-quote-summary/main-quote-summary.stories.js:66]: I think you will get a lint error here

```suggestion
      <MainQuoteSummary {...args} />
```

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Fantastic work! 2 tiny changes and I can approve. Thanks!

**Comment by Jordon-L** [comment]
I have applied the changes. Thank you for the reviews.

**Review by georgewrmarshall — DISMISSED** [review]
Excellent! 💯 Thanks for your contribution 🙏

**Review by darkwing — APPROVED** [review]

**Review by darkwing — CHANGES_REQUESTED** [review]
Ooops @Jordon-L , there are lint issues. Try `yarn lint:fix` to get those cleaned up!

**Review by darkwing — DISMISSED** [review]

**Review by georgewrmarshall — DISMISSED** [review]

**Comment by Jordon-L** [comment]
Ok, thanks for letting me know.

**Review by brad-decker — APPROVED** [review]

**Review by darkwing — APPROVED** [review]

**Pull request merged**